chatrooms.lua

```diff
@@ -29,12 +35,22 @@ function OnStartup()
 				sDescription = "Chatroom for usage by VIPs.",
 				sEmail = "donot@mail.me"
 			},
-			sFileName = "VIPChat.txt"
-		}
+			sFileName = "VIPChat.txt",
+			tChatHistory = {},
+			tUsers = {},
+		},
 	}
-	for sIndex, sValue in pairs(tChatRooms) do
-		Core.RegBot( sIndex, sValue.BOT.sDescription, sValue.BOT.sEmail, true )
+	for sIndex, tValue in pairs( tChatRooms ) do
+		Core.RegBot( sIndex, tValue.BOT.sDescription, tValue.BOT.sEmail, true )
 	end
+	dofile( sPath..sExtPath.."offliner.lua" )
```

Comment:
1. `offliner.lua` relies heavily on luasql connection to MySQL. You should also note that [`SQLCon`][1] was never created in this context.
1. From the definition of [`SendToRoom`][2], you can see that we are not interested in offline users (and even if we are, you can maintain a table).
1. The whole list of moderators has over 500 users. The `!sm` command only shows active moderators. In the chatroom though, we only need to show currently online users.
1. Registered users with VIPs or above privileges will be listed for `#[VIPChat]` as well as `#[ModzChat]`. Keep only the set difference in the second table.
  [1]: http://git.io/ALCi
  [2]: https://github.com/nishithshah2211/ptokax-scripts/blob/chathistory/chatrooms.lua#L69